### ISSUE TITLE
Upgrade dependencies

### ISSUE THREAD

**Issue by alex-ber** [alex-ber/AlexBerUtils#6]
As part of preparation of migrating to Python 3.8 I want to upgrade some dependencies.
Also pytest in use is old, so I will upgrade to the latest version. 

Note, that all req-*.txt remains unchanged. These files are used in setup.py.
requirements-*.txt file are used to create reproducible environment, so if you are not using them directly their changes will not effect you.

**Comment by alex-ber** [comment]
```
req-env (python-dotenv>=0.13.0): 
   is unchanched
requirements-env.txt (python-dotenv==0.15.0):
   python-dotenv version changed from 0.13.0 to 0.15.0 - to the latest, has only minor changes, see 
             https://github.com/theskumar/python-dotenv/compare/v0.13.0...v0.15.0

requirements.txt (PyYAML==5.1)
   is unchanched

req-fabric.txt (fabric>=2.5.0)
   is unchanched

requirements-fabric.txt  (fabric==2.5.0)
  asn1crypto removed
  bcrypt upgraded from 3.1.6 to 3.2.0
  cffi upgraded from 1.11.5 to 1.14.3 as preparation for supporting Python 3.8. v1.12.3 has "More 3.8 fixes"
  see https://cffi.readthedocs.io/en/latest/whatsnew.html#older-versions
  cryptography upgraded from 2.6.1 to 3.1.1
  paramiko upgraded from 2.7.1 to 2.7.2 - some bugs were fixed, see https://www.paramiko.org/changelog.html
  pycparser upgraded from 2.19 to 2,20
  six upgraded from 1.12.0 to 1.15.0

req-yml.txt
   is unchanched

requirements-yml.txt
 MarkupSafe is upgraded from 1.1.0 to 1.1.1, see https://markupsafe.palletsprojects.com/en/1.1.x/changes/#version-1-1-1
 Jinja2 is upgraded from 2.10.1 to 2.11.2, v 2.10.2 has better support of Python 3.7, see https://jinja.palletsprojects.com/en/2.11.x/changelog/

req-md.txt  (multidispatch>=0.2)
   is unchanched

requirements-md.txt (multidispatch==0.2)
   is unchanched, no new version was released.

requirements-tests.txt
  see below.

```

**Comment by alex-ber** [comment]
Upgrade test dependencies:

```
requirements-tests.txt
   pytest upgraded from 4.3.1 to the latest 6.1.2. Meanwhile, it's dependencies was also changed, so I:
     removed 
       atomicwrites (not required).
       six (not required)
     added
       toml==0.10.2
       pluggy==0.13.1
       packaging==20.4
       iniconfig==1.1.1
       pyparsing==2.4.7

   pytest-mock was upgraded from 1.10.1 to 3.3.1 (latest). Previous version didn't work with latest pytest.
   mock was upgraded from 2.0.0 to 4.0.2 (latest) to accompaniment pytest-mock.
   pytest-assume was upgraded from 1.2.2 to 2.3.3 (latest) Previous version didn't work with latest pytest. 
   
    attrs was upgraded from 19.1.0 to 20.2.0 (at least 19.2.0 is required by Twisted, so upgrade is needed)
    py was upgraded from 1.8.0. to 1.9.0
 ```

**Comment by alex-ber** [comment]
Released in 0.6.0a4.
